src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,54 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private int salaryPerHour;
+    private int workingHours;
+    private LocalDate localDateFrom;
+    private LocalDate localDateTo;
+    private String[] splitData;
+    private int[] salaries;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        localDateFrom = toLocalDate(dateFrom);
+        localDateTo = toLocalDate(dateTo);
+        salaries = new int[names.length];
+        for (int i = 0; i < data.length; i++) {
+            splitData = data[i].split(" ");
+            workingHours = Integer.parseInt(splitData[2]);
+            salaryPerHour = Integer.parseInt(splitData[3]);
+            LocalDate date = toLocalDate(splitData[0]);
+            if (date.isAfter(localDateFrom)
+                    && date.isBefore(localDateTo)
+                    || date.isEqual(localDateFrom)
+                    || date.isEqual(localDateTo)) {
+                for (int z = 0; z < names.length; z++) {
+                    if (names[z].equals(splitData[1])) {
+                        salaries[z] += workingHours * salaryPerHour;
+                    }
+                }
+            }
+        }
+        return buildReport(names, salaries, dateFrom, dateTo);
+    }
+
+    private LocalDate toLocalDate(String date) {
+        return LocalDate.parse(date, DateTimeFormatter.ofPattern("dd.MM.yyyy"));
```

Comment:
[checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-core/exceptions/salary-info): If you create a [formatter](https://docs.oracle.com/javase/tutorial/datetime/iso/format.html), make it a constant field.